libs/shared/feature/schema-nuv/src/lib/components/forms/header/schema-builder-header.component.ts

```diff
@@ -1,17 +1,101 @@
 import { CommonModule } from '@angular/common';
-import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { MatDialog, MatDialogModule, MatDialogRef } from '@angular/material/dialog';
 import { ISchemaMetaData } from '@dsg/shared/data-access/work-api';
-import { NuverialButtonComponent, NuverialIconComponent, NuverialSpinnerComponent } from '@dsg/shared/ui/nuverial';
+import { UserStateService } from '@dsg/shared/feature/app-state';
+import { NuverialButtonComponent, NuverialIconComponent, NuverialSnackBarService, NuverialSpinnerComponent } from '@dsg/shared/ui/nuverial';
+import { EMPTY, catchError, filter, firstValueFrom, switchMap, take, tap } from 'rxjs';
+import { SchemaBuilderService } from '../builder/schema-builder.service';
+import { SchemaDefinitionMetaDataComponent } from '../schema-definition-metadata/schema-definition-metadata.component';
 
 @Component({
   changeDetection: ChangeDetectionStrategy.OnPush,
-  imports: [CommonModule, NuverialSpinnerComponent, NuverialIconComponent, NuverialButtonComponent],
+  imports: [CommonModule, MatDialogModule, NuverialSpinnerComponent, NuverialIconComponent, NuverialButtonComponent],
   selector: 'dsg-schema-builder-header',
   standalone: true,
   styleUrls: ['./schema-builder-header.component.scss'],
   templateUrl: './schema-builder-header.component.html',
 })
-export class BuilderHeaderComponent {
+export class SchemaBuilderHeaderComponent implements OnChanges {
+  constructor(
+    private readonly _dialog: MatDialog,
+    private readonly _cdr: ChangeDetectorRef,
+    protected readonly _userStateService: UserStateService,
+    protected readonly _schemaBuilderService: SchemaBuilderService,
+    protected readonly _nuverialSnackBarService: NuverialSnackBarService,
+  ) {
+    this.lastUpdatedDisplay = '';
+    this.createdByDisplay = '';
+  }
+
   @Input() public metaData?: ISchemaMetaData | null;
+  public lastUpdatedDisplay = '';
+  public createdByDisplay = '';
   public loading = false;
+  public status = '';
+  public dialogRef?: MatDialogRef<SchemaDefinitionMetaDataComponent>;
+  public async ngOnChanges(changes: SimpleChanges): Promise<void> {
+    if (changes['metaData']?.currentValue && this.metaData) {
+      await this.updateLastUpdatedBy();
+      await this.updateCreatedBy();
+      this._cdr.detectChanges();
+    }
+  }
+
+  public async updateLastUpdatedBy(): Promise<void> {
+    if (this.metaData) {
+      const lastUpdatedByUser = await firstValueFrom(this._userStateService.getUserById$(this.metaData.lastUpdatedBy ?? ''));
+      if (lastUpdatedByUser?.displayName && this.metaData.lastUpdatedBy !== lastUpdatedByUser.displayName) {
+        this.lastUpdatedDisplay = lastUpdatedByUser.displayName;
+        this._cdr.detectChanges();
+      }
+    }
+  }
```

Comment:
Do you just need the username here can you use `getUserDisplayName$`?

```suggestion
  public async updateLastUpdatedBy(): Promise<void> {
    if (this.metaData) {
      this.lastUpdatedDisplay = await firstValueFrom(getUserDisplayName$);
        this._cdr.detectChanges();
    }
  }
```